libraries/triedent/src/cache_allocator.cpp

```diff
@@ -21,12 +21,20 @@ namespace triedent
              [this]()
              {
                 thread_name("swap");
+#ifndef __APPLE__
                 pthread_setname_np(pthread_self(), "swap");
+#else // if __APPLE__
+                pthread_setname_np("swap");
+#endif
```

Comment:
Can we wrap setting the thread name in a function, so that this #ifdef isn't duplicated all over?

Reply:
Wrapped in function set_current_thread_name()